manifests/pg_repack.pp

```diff
@@ -0,0 +1,137 @@
+# Maintenance pg_repack
+#
+# @summary Maintenance pg_repack
+
+class pe_databases::pg_repack (
+  Boolean $disable_maintenance = false,
+  Integer $jobs                = $facts['processors']['count'] / 4
+) {
+
+  $ensure_service = $disable_maintenance ? {
+    true  => stopped,
+    default => running,
+  }
+
+  $ensure_file = $disable_maintenance ? {
+    true    => absent,
+    default => present
+  }
+
+  # PE 2019.1 starting shipping versioned pe-postgres packages where all paths are versioned.
+  # So, prior to 2019.1 use a non-versioned path, and after use a versioned path.
+  # TODO: Use $pe_databases::psql_version after identifying why it is cast to ${psql_version}00000 in spec tests.
+  $postgresql_version = $facts['pe_postgresql_info']['installed_server_version']
+  $repack_executable = versioncmp('2019.1.0', $facts['pe_server_version']) ? {
+                          1       => '/opt/puppetlabs/server/apps/postgresql/bin/pg_repack',
+                          default => "/opt/puppetlabs/server/apps/postgresql/${postgresql_version}/bin/pg_repack"
+                          }
+
+  $repack          = "su - pe-postgres -s /bin/bash -c \"${repack_executable} -d pe-puppetdb"
+  $repack_jobs     = "--jobs ${jobs}"
+
+  $facts_tables    = '-t factsets -t fact_paths"'
+  $catalogs_tables = versioncmp($facts['pe_server_version'], '2019.8.1') ? {
+                      1       => '-t catalogs -t catalog_resources -t catalog_inputs -t edges -t certnames"',
+                      default => '-t catalogs -t catalog_resources -t edges -t certnames"' }
+  $other_tables    = '-t producers -t resource_params -t resource_params_cache"'
+  $reports_table   = '-t reports"'
+  $resource_events_table = '-t resource_events"'
+
+  Service {
+    notify  => Exec['pe_databases_daemon_reload'],
+  }
+  File {
+    notify  => Exec['pe_databases_daemon_reload'],
+  }
+
+  file { '/etc/systemd/system/pe_databases-facts.service':
```

Comment:
We might as well as make a defined type for these.

Reply:
:+1: